powerauth-java-crypto/src/test/java/io/getlime/security/powerauth/crypto/signature/PowerAuthSignatureTest.java

```diff
@@ -315,7 +320,141 @@ public void testSignatureForDataV3() throws Exception {
                 System.out.println("### Server Signature Key: " + BaseEncoding.base64().encode(keyConvertor.convertSharedSecretKeyToBytes(signatureServerKeyKnowledge)));
                 assertEquals(signatureClientKeyKnowledge, signatureServerKeyKnowledge);
 
-                boolean isSignatureValid = serverSignature.verifySignatureForData(data, signature, Arrays.asList(signatureServerKeyPossession, signatureClientKeyKnowledge), ctrData);
+                boolean isSignatureValid = serverSignature.verifySignatureForData(data, signature, Arrays.asList(signatureServerKeyPossession, signatureClientKeyKnowledge), ctrData, signatureFormat);
+                System.out.println("## Signature valid: " + (isSignatureValid ? "TRUE" : "FALSE"));
+                assertTrue(isSignatureValid);
+
+                ctrData = ctrGenerator.next(ctrData);
+            }
+        }
+    }
+
+    /**
+     * Test of signature generation and validation.
+     *
+     * <p><b>PowerAuth protocol versions:</b>
+     * <ul>
+     *     <li>3.1</li>
+     * </ul>
+     *
+     * @throws java.lang.Exception If the test fails.
+     */
+    @Test
+    public void testSignatureForDataV31() throws Exception {
+        System.out.println("# PowerAuth Signature");
+        System.out.println();
+
+        // Prepare data
+        KeyGenerator keyGenerator = new KeyGenerator();
+        CryptoProviderUtil keyConvertor = PowerAuthConfiguration.INSTANCE.getKeyConvertor();
+
+        KeyPair serverKeyPair = keyGenerator.generateKeyPair();
+        PrivateKey serverPrivateKey = serverKeyPair.getPrivate();
+        PublicKey serverPublicKey = serverKeyPair.getPublic();
+
+        System.out.println("## Server Private Key: " + BaseEncoding.base64().encode(keyConvertor.convertPrivateKeyToBytes(serverPrivateKey)));
+        System.out.println("## Server Public Key:  " + BaseEncoding.base64().encode(keyConvertor.convertPublicKeyToBytes(serverPublicKey)));
+
+        KeyPair deviceKeyPair = keyGenerator.generateKeyPair();
+        PrivateKey devicePrivateKey = deviceKeyPair.getPrivate();
+        PublicKey devicePublicKey = deviceKeyPair.getPublic();
+
+        System.out.println("## Device Private Key: " + BaseEncoding.base64().encode(keyConvertor.convertPrivateKeyToBytes(devicePrivateKey)));
+        System.out.println("## Device Public Key:  " + BaseEncoding.base64().encode(keyConvertor.convertPublicKeyToBytes(devicePublicKey)));
+
+        final PowerAuthSignatureFormat signatureFormat = PowerAuthSignatureFormat.getFormatForSignatureVersion("3.1");
+        assertEquals(signatureFormat, PowerAuthSignatureFormat.BASE64);
```

Comment:
This is slightly incorrect. Expected value is first parameter in `assertEquals` method, so we should swap the parameters.